modules/_labs/button/react/lib/types.ts

```diff
@@ -0,0 +1,90 @@
+import {CSSObject} from '@emotion/core';
+
+/**
+ * Standard button Button
+ */
+export enum ButtonVariant {
+  Primary = 'primary',
+  Secondary = 'secondary',
+  OutlinePrimary = 'outlinePrimary',
+  OutlineSecondary = 'outlineSecondary',
+  OutlineInverse = 'outlineInverse',
+}
+export enum ButtonSize {
+  Small = 'small',
+  Medium = 'medium',
+  Large = 'large',
+}
+export enum ButtonIconPosition {
+  Left = 'iconPositionLeft',
+  Right = 'iconPositionRight',
+}
+
+/**
+ * Dropdown Button
+ */
+export enum DropdownButtonVariant {
+  Primary = 'primary',
+  Secondary = 'secondary',
+}
+
+/**
+ * Icon Button
+ */
+export enum IconButtonVariant {
+  Square = 'square',
+  SquareFilled = 'squareFilled',
+  Plain = 'plain',
+  Circle = 'circle',
+  CircleFilled = 'circleFilled',
+  Inverse = 'inverse',
+  InverseFilled = 'inverseFilled',
+}
+
+/**
+ * Text Button
+ */
+export enum TextButtonVariant {
+  Default = 'text',
+  Inverse = 'textInverse',
+  AllCaps = 'textAllCaps',
+  InverseAllCaps = 'textInverseAllCaps',
+}
+
+/**
+ * Old orange buttons
+ * @deprecated These type are deprecated along with deprecated_Button component
+ */
+export enum DeprecatedButtonVariant {
+  Primary = 'deprecatedPrimary',
+  Secondary = 'deprecatedSecondary',
+  Delete = 'deprecatedDelete',
+}
+
+export type AllButtonVariants =
+  | ButtonVariant
+  | DropdownButtonVariant
+  | TextButtonVariant
+  | IconButtonVariant
+  | DeprecatedButtonVariant;
+
+/**
+ * The object used for passing in colors to the ButtonContainer
+ */
+export interface ButtonStateColors {
+  background?: string;
+  border?: string;
+  icon?: string;
+  iconFill?: boolean;
+  label?: string;
+  labelData?: string;
```

Comment:
`ButtonStateColors` makes me think of what you have in `ButtonColors` is there a reason we named it states here?

Reply:
@mannycarrera4 These are the colors for a single state, `ButtonColors` are the entire set of colors for all states